src/explore-education-statistics-common/src/components/ChartsToPNG.tsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useExportButtonContext } from '@common/contexts/ExportButtonContext';
+import html2canvas, { Options as HTML2CanvasOptions } from 'html2canvas';
+import { useCallback, useState } from 'react';
+
+export type UseGenerateImage<T extends HTMLElement = HTMLDivElement> = [
+  (callback?: BlobCallback) => Promise<string | undefined>,
+  {
+    isLoading: boolean;
+  },
+];
+
+export type UseGenerateImageArgs = {
+  options?: HTML2CanvasOptions;
+  quality?: number;
+  type?: string;
+};
+
+export function useGenerateImage(
+  args?: UseGenerateImageArgs,
+): UseGenerateImage {
+  const [isLoading, setIsLoading] = useState(false);
+  const ref = useExportButtonContext();
+
+  const generateImage = useCallback(
+    // eslint-disable-next-line consistent-return
+    async (callback?: BlobCallback) => {
+      if (ref && ref.current) {
+        setIsLoading(true);
+
+        return html2canvas(ref.current, {
+          logging: false,
+          ...args?.options,
+        }).then(canvas => {
+          if (callback) {
+            canvas.toBlob(callback, args?.type, args?.quality);
+          }
+          setIsLoading(false);
+          return canvas.toDataURL(args?.type, args?.quality);
+        });
+      }
+    },
+    [args, ref],
+  );
+
+  return [
+    generateImage,
+    {
+      isLoading,
+    },
+  ];
+}
+
+export type UseCurrentPng = [
+  (callback?: BlobCallback) => Promise<string | undefined>,
+  {
+    isLoading: boolean;
+    ref: React.MutableRefObject<any>;
```

Comment:
I think this can be `React.MutableRefObject<HTMLDivElement | null>;` so we don't need to use `any`